lib/schnorr.py

```diff
@@ -0,0 +1,110 @@
+import os
+import sys
+from . import secp256k1
+from ctypes import (create_string_buffer, memset, c_void_p, c_char_p, c_int,
+                    c_size_t)
+
+class SecpMissing(RuntimeError):
+    ''' Base class of the below two exception classes '''
+
+class SecpMissingLibrary(SecpMissing):
+    ''' Raised to indicate we are missing libsecp256k1 '''
+
+class SecpMissingSchnorr(SecpMissing):
+    ''' Raised to indicate we are missing the requisite Schnorr functions from
+    libsecp256k1, even though we have libsecp256k1 '''
+
+def _boilerplate_secp_check(which):
+    if not secp256k1.secp256k1:
+        raise SecpMissingLibrary('Missing library libsecp256k1, or the library could not be loaded')
```

Comment:
These should be translatable

Reply:
I would normally agree with you but I think these exceptions won't ever make it to the UI. They are there for future programmer defense.

The way I envision it is the UI options for Schnorr will be grayed out.  I'm force-pushing a commit now that offers `is_available` and all UI flows will first check with that function before proceeding.